lib/rules/no-implied-eval.js

```diff
@@ -12,51 +12,77 @@
 //------------------------------------------------------------------------------
 
 module.exports = function(context) {
-    var IMPLIED_EVAL = /set(?:Timeout|Interval)|execScript/;
+    var CALLEE_RE = /set(?:Timeout|Interval)|execScript/;
+
+    // Figures out if we should inspect a given binary expression. Is a stack of
+    // stacks, where the first element in each substack is a CallExpression.
+    var dangerousAncestorsStack = [];
 
     //--------------------------------------------------------------------------
     // Helpers
     //--------------------------------------------------------------------------
 
     /**
-     * Checks if the first argument of a given CallExpression node is a string literal.
-     * @param {ASTNode} node The CallExpression node the check.
-     * @returns {boolean} True if the first argument is a string literal, false if not.
+     * Get the last element of an array, without modifying arr, like pop(), but non-destructive.
+     * @param {array} arr What to inspect
+     * @returns {*} The last element of arr
+     * @private
      */
-    function hasStringLiteralArgument(node) {
-        var firstArgument = node.arguments[0];
-
-        return firstArgument && firstArgument.type === "Literal" && typeof firstArgument.value === "string";
+    function last(arr) {
+        return arr ? arr[arr.length - 1] : null;
     }
 
     /**
-     * Checks if the given MemberExpression node is window.setTimeout or window.setInterval.
+     * Checks if the given MemberExpression node is a dangerous identifier on window.
      * @param {ASTNode} node The MemberExpression node to check.
-     * @returns {boolean} Whether or not the given node is window.set*.
+     * @returns {boolean} Whether or not the given node is potentially dangerous.
+     * @private
      */
-    function isSetMemberExpression(node) {
+    function isDangerousMemberExpression(node) {
```

Comment:
I think the term "dangerous" is confusing throughout. I'd replace with "implied eval" wherever possible.